contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -33,23 +33,26 @@ contract OrderValidationUtils is
     using LibBytes for bytes;
 
     // solhint-disable var-name-mixedcase
-    IExchange internal EXCHANGE;
-    bytes internal ZRX_ASSET_DATA;
-    address internal ERC20_PROXY_ADDRESS;
+    bytes internal _ZRX_ASSET_DATA;
     // solhint-enable var-name-mixedcase
 
     constructor (address _exchange, bytes memory _zrxAssetData)
         public
+        LibAssetData(_exchange)
     {
-        EXCHANGE = IExchange(_exchange);
-        ZRX_ASSET_DATA = _zrxAssetData;
-        ERC20_PROXY_ADDRESS = EXCHANGE.getAssetProxy(ERC20_PROXY_ID);
+        _ZRX_ASSET_DATA = _zrxAssetData;
     }
 
-    /// @dev Fetches information for order and maker/taker of order.
+    /// @dev Fetches all order-relevant information needed to validate if the supplied order is fillable.
     /// @param order The order structure.
-    /// @param signature Proof that order has been created by maker.
-    /// @return OrderInfo, the remaining amount fillable by the taker, and validity of signature for given order.
+    /// @param signature Signature provided by maker that proves the order's authenticity.
+    /// `0x01` can always be provided if the signature does not need to be validated.
+    /// @return orderInfo The hash, status, and `takerAssetAmount` already filled for the given order.
+    /// @return fillableTakerAssetAmount The amount of the order's `takerAssetAmount` that is fillable given all on-chain state.
+    /// If the `takerAssetData` encodes data for multiple assets, `fillableTakerAssetAmount` will represent a "scaled"
+    /// amount, meaning it must be multiplied by all the individual asset amounts within the `takerAssetData` to get the final
+    /// amount of each asset that can be filled.
+    /// @return isValidSignature The validity of the provided signature.
```

Comment:
My gut is multiple `@return` won't work. Can we either remove or find out?

Reply:
It didn't work well, removed.